### PR TITLE
Added support for custom dyna beans

### PR DESCRIPTION
**Pull request by elvinluis29**
Hi, first I want to thank you for these great tools. I was wondering if you can merge these changes. I am working in a project where the properties of the container change dynamically depending on some conditions, the underlying objects I am using are JSON objects which I wrap with an implementation of DynaBean. The issue I noticed in your container is that my JSONDynaBean was wrapped in your code by the WrapDynaBean, even though it does not cause any problem I thought it could be improved to handle this situation.

Thanks

### CODE REVIEW

File: src/main/java/org/vaadin/viritin/ListContainer.java [hunk 1]
```diff
@@ -595,8 +609,13 @@ public Class getType() {
...
             public boolean isReadOnly() {
-                return getDynaClass().getPropertyDescriptor(propertyName).
-                        getWriteMethod() == null;
+            	DynaClass clazz = getDynaClass();
+            	if(clazz instanceof WrapDynaClass){
+            		return ((WrapDynaClass)clazz).getPropertyDescriptor(propertyName).
+            				getWriteMethod() == null;
+            	}
+                
+            	return false;
```
Comment (mstahv) [inline src/main/java/org/vaadin/viritin/ListContainer.java]: Read only unsupported in all custom cases?

File: src/main/java/org/vaadin/viritin/ListContainer.java [hunk 2]
```diff
@@ -621,7 +640,11 @@ public T getBean() {
         private DynaBean getDynaBean() {
             if (db == null) {
                 try {
-                    db = new WrapDynaBean(bean, getDynaClass(bean));
+                	if(bean instanceof DynaBean){
+                		db = (DynaBean) bean;
+                	}else{
```
Comment (mstahv) [inline src/main/java/org/vaadin/viritin/ListContainer.java]: Add spaces around else word

Comment (elvinluis29) [inline src/main/java/org/vaadin/viritin/ListContainer.java]: Added spaces around else statement

**Comment by elvinluis29** [comment]
No, I did not run any performance test, but I think it should not be a huge difference.

I'll try to upload some usage example later.

**Comment by elvinluis29** [comment]
I wrote a usage example here: https://github.com/elvinluis29/viritin/wiki

In summary, in the example I try to create a database table browser. Let me know if you have any questions.

**Comment by mstahv** [comment]
Created a demo app showing this kind of DynaBean usage here:
https://github.com/mstahv/dynamic-database-browser-example

**Comment by elvinluis29** [comment]
Great! Thanks.

**Pull request merged**